Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/DBRequestClient.java

```diff
@@ -42,11 +47,74 @@ public ReviewRequest getReviewRequestForConcept(String conceptId){
         request.setConceptId(conceptId);
         
         List<ReviewRequest> reviewRequests = client.queryByExample(request);
-        
         if(reviewRequests!= null && reviewRequests.size()>0) {
-            return reviewRequests.get(0);
+            return reviewRequests.get(reviewRequests.size() - 1);
         }
+        
         return null;
     }
-   
+    
+    @Override
+    public List<ReviewRequest> getAllReviewsForaConcept(String conceptId){
+        ReviewRequest request = new ReviewRequest();
+        request.setConceptId(conceptId);
+        
+        List<ReviewRequest> reviewRequests = client.queryByExample(request);
+        
+        return reviewRequests;
+    }
+    
+    @Override
+    public ReviewRequest updateReviewRequest(ReviewRequest reviewRequest) {
+        
+        List<ReviewRequest> requests = client.query(new Predicate<ReviewRequest>() {
+            private static final long serialVersionUID = 6495914730735826451L;
+
+            @Override
+            public boolean match(ReviewRequest review) {
+                return review.getConceptId().equals(reviewRequest.getConceptId());
+            }
+            
+        });
+       
+        List<String> comments  = requests.get(requests.size() - 1).getResolvingComment();
+        comments.addAll(reviewRequest.getResolvingComment());
+        
+        requests.get(requests.size() - 1).setResolvingComment(comments);
+        requests.get(requests.size() - 1).setStatus(reviewRequest.getStatus());
+        requests.get(requests.size() - 1).setResolver(reviewRequest.getResolver());
```

Comment:
This I'm not clear. We need to update the already present data in the database. If we are updating the data in the manager and calling the store in here will add duplicate rows right ?

Reply:
that is probably true for db4o but shouldn't for hibernate. So once we switch, the manager should only make an update call to the db class. For hibernate you might have to do the retrieve and copy, but that should be a general update method (not specific to this usecase).

Reply:
Yeah. Then the current flow should work right. I have the updated request which I'm sending from Manager. Because of db4o, I'm retrieving the required data to be updated and I'm calling store. In case of Hibernate, we should be able to add directly after include a unique id parameter

Reply:
I think the issue, which I have with all of this is probably related more to the data model which bleeds into the storing procedure here. Each comment will have a creator and a creation date (which they don't have here, since they are just strings). Once the comments are proper objects, you can directly store them in the database and update modified requests.

Reply:
Ah Yeah, I'm adding that as well. But even if I add those attributes I think we need to follow how we did before only in case of db4o. We need to somehow point to the place where the last ReviewRequest of given concept. Upon finding that we need to perform set() operation.

Reply:
The user should see the last request (by creation date) and then you have an id for that request. There would be no list involved, just retrieve request by id and resolve it/add comment.